### PR TITLE
Completion for magick

### LINKED ISSUE #7172: Update ImageMagick completions
In ImageMagick 7 or later, legacy commands have been replaced with `magick`. The current completions are only valid if there is a symbolic links for compatibility.
Therefore, I think it would be better to update completions to support `magick` and it's sub-commands for compatibility.

See: [Command-line Tools](https://imagemagick.org/script/command-line-tools.php) and [Porting to ImageMagick Version 7](https://imagemagick.org/script/porting.php)

### PR DESCRIPTION
**Pull request by jadijadi**
In ImageMagick 7 or later, legacy commands have been replaced with magick. Here a new file is added for the magick command. It is based on the `convert.fish` but compared against `man magick` for fixes.

fixes #7172

### CODE REVIEW

File: share/completions/magick.fish [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+complete -c magick -o type -d 'Image type [type]' -xa '(convert -list type)'
+complete -c magick -o unique-colors -d 'Discard all but one of any pixel color'
+complete -c magick -o unsharp -d 'Sharpen the image [geometry]'
+complete -c magick -o vignette -d 'Soften the edges of the image in vignette style [geometry]'
+complete -c magick -o wave -d 'Alter an image along a sine wave [geometry]'
+complete -c magick -o white-threshold -d 'Force all pixels above the threshold into white [value]'
+complete -c magick -o channel-extract -d 'extract the channels in order'
+complete -c magick -o channel-inject -d 'inject the channels in order'
+complete -c magick -o channel-swap -d 'swap the channels'
+complete -c magick -o affinity 'transform image colors to match this set of colors'
```
Comment (mqudsi) [inline share/completions/magick.fish:192]: Missing `-d` here

Comment (jadijadi) [reply share/completions/magick.fish:192]: Thanks for the missing `-d` catch. Added it and pushed -ff again.

**Comment by jadijadi** [comment]
> Thanks for this.
> 
> It would be ideal to share the same completions for `magick convert` and `convert`, if possible.

how? I was not sure if they should be 100% same so did not touched the convert.

File: share/completions/magick.fish (hunk 1 above)
Comment (jadijadi) [reply share/completions/magick.fish]: Ouch . sorry; my bad. have pushed to the wrong branch on my fork. re-pushed.

**Comment by krobelus** [comment]
> It would be ideal to share the same completions for magick convert and convert, if possible.
> 
> how? I was not sure if they should be 100% same so did not touched the convert.

create a `share/functions/__fish_complete_magick.fis` fish all the common options:

```fish
function __fish_complete_magick
	complete $argv -o adjoin -d 'Join images into a single multi-image file'
	...
end
```

and call it as `__fish_complete_magick magick` or `__fish_complete_magick convert`.

**Comment by jadijadi** [comment]
> ```fish
> complete $argv -o adjoin -d 'Join images into a single mu
> ```

Thanks for the hint. Clean and clear. will update my PR soon

**Review by mqudsi — APPROVED** [review]

**Comment by mqudsi** [comment]
Merged in 6840ef46f794618c3b5bf053247b4e9111d4adc6

Thank you very much.

**Pull request closed without merging**